app/src/main/java/com/example/myapplication/ui/MainActivityCompose.kt

```diff
@@ -0,0 +1,133 @@
+package com.example.myapplication.ui
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.GridItemSpan
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color.Companion.Black
+import androidx.compose.ui.graphics.Color.Companion.Blue
+import androidx.compose.ui.graphics.Color.Companion.Cyan
+import androidx.compose.ui.graphics.Color.Companion.Gray
+import androidx.compose.ui.graphics.Color.Companion.Magenta
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class MainActivityCompose : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            val mainViewModel: MainActivityViewModel = viewModel()
+
+            FlexibleGird(
+                itemsCount = 10,
+                columnsCount = 3,
+                itemsHeight = 100,
+                gridWidth = 300,
+                itemsHeightPercent = 20,
+                parentPaddingStart = 15,
+                parentPaddingEnd = 15,
+                parentPaddingTop = 15,
+                parentPaddingBottom = 15,
+                itemDecorationVertical = 15,
+                itemDecorationHorizontal = 15,
+                hasDinamycHeight = false
+            )
+        }
+    }
+}
+
+@Preview(showBackground = true)
+@Composable
+fun DefaultPreview() {
+    FlexibleGird(
+        itemsCount = 10,
+        columnsCount = 3,
+        itemsHeight = 100,
+        gridWidth = 300,
```

Comment:
gridWidth better set via modifier and do it on upper-level function:
Here we need set `val modifier: Modifier = Modifier` as first field for FlexibleGird fields
and in upper-level component we can just set our parameters via Modifier:
```suggestion
    FlexibleGrid(
        modifier = Modifier.width(300.dp)
        ...
    )
```
Then, on 70 line of this file you need just pass modifier to LazyVerticalGrid modifier:
```suggestion
        LazyVerticalGrid(
            modifier = modifier,
            ...
        )
```